src/dmd/mars.d

```diff
@@ -1508,6 +1508,20 @@ private bool parseCommandLine(const ref Strings arguments, const size_t argc, re
             params.useDeprecated = 2;
         else if (arg == "-c")                // https://dlang.org/dmd.html#switch-c
             params.link = false;
+        else if (startsWith(p + 1, "checkaction=")) // https://dlang.org/dmd.html#switch-checkaction
+        {
+            /* Parse:
+             *    -checkaction=D|C|halt
+             */
+            if (strcmp(p + 13, "D") == 0)
```

Comment:
Perhaps a case insensitive comparison.

Reply:
We don't do it for any other switches.

Reply:
We don't have any other flag values that are capital letters.

Reply:
Nobody refers to D or C as d or c.

Reply:
Nobody uses capital letters in command line flags 😉.

Reply:
I would err on the side of @jacob-carlborg here, all cross language tools I can think off the top of my head refer to languages in lower case on the command-line.

E.g
```
gcc -x c++ foo.cc -x c bar.c
c++filt -s java
highlight -S c
```

But I don't think I should dictate what dmd should do, as I can do whatever I feel is appropriate for gdc command-line options anyway.

Reply:
Same with Clang.